vcpkg.json

```diff
@@ -0,0 +1,26 @@
+{
+    "name": "tiledb",
```

Comment:
```suggestion
    "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
    "name": "tiledb",
    "description": "The TileDB Embedded Universal Storage Engine",
    "license": "MIT",
    "homepage": "https://tiledb.com/products/tiledb-embedded/",
    "documentation": "https://docs.tiledb.com/main/",
```

Reply:
Also, should we change the name to libtiledb here?

Reply:
Few other vcpkg ports have a `lib` prefix and it seems redundant IMHO.

Reply:
That's a fair point. Though I was more worried if we (the TileDB we) end up with a separate vcpkg port that would be named "tiledb" that isn't what we normally think of as `libtiledb.{so,dylib,dll}`.

Reply:
Revisit if/when we upstream